pkg/controllers/termination/controller.go

```diff
@@ -37,25 +39,52 @@ import (
 
 // Controller for the resource
 type Controller struct {
-	terminator    *Terminator
-	cloudProvider cloudprovider.CloudProvider
-	kubeClient    client.Client
+	Terminator *Terminator
+	KubeClient client.Client
+}
+
+// For returns the resource this controller is for.
+func (c *Controller) For() client.Object {
+	return &v1.Node{}
+}
+
+func (c *Controller) Name() string {
+	return "terminator"
 }
 
 // NewController constructs a controller instance
 func NewController(kubeClient client.Client, coreV1Client corev1.CoreV1Interface, cloudProvider cloudprovider.CloudProvider) *Controller {
 	return &Controller{
-		terminator:    &Terminator{kubeClient: kubeClient, cloudProvider: cloudProvider, coreV1Client: coreV1Client},
-		cloudProvider: cloudProvider,
-		kubeClient:    kubeClient,
+		KubeClient: kubeClient,
+		Terminator: &Terminator{kubeClient: kubeClient, cloudProvider: cloudProvider, coreV1Client: coreV1Client,
+			evictionQueue: EvictionQueue{
+				queue:        workqueue.NewRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(100*time.Millisecond, 10*time.Second)),
```

Comment:
Can you extract and name these constants?

Reply:
Good idea. Will do.